### PR TITLE
Added a modular system for bypassing different linkvertise content types and added a paste bypass.

### PR DESCRIPTION
**Pull request by davidkra230**
\#### continuation of #928 and #933, third time's the charm. (next time I should really only push when I'm 100% done) ####

Fixes (Links to issues fixed by this PR): 
None

Description:
bypasses linkvertise's pasting feature. basically i just added a check for pastes and if it matches it will get the info almost exactly like the url bypass ~~except it copies the paste instead of redirecting~~ but it redirects to a blob url of the paste text.

Test links:
[LinkvertisePaste](https://linkvertise.com/114822/test?o=sharing)
Checklist:

- [x] I made sure there are no unnecessary changes in the code*
- [x] Tested on Chromium- Browser OS
- [x] Tested on Firefox

### CODE REVIEW

File: src/js/injection_script.js [hunk 1]
```diff
@@ -2850,31 +2850,65 @@ ensureDomLoaded(() => {
...
+                                                "Accept": 'application/json',
+                                                "Content-Type": 'application/json'
+                                            }
+                                        }).then(r=>r.json()).then(json=>{
+                                            if (json?.data.paste) {
+                                                insertInfoBox('Bypassed paste! Opening in 3 seconds..')
+                                                setTimeout(() => {
+                                                    window.location.href =
+                                                    URL.createObjectURL(
+                                                        new Blob([json?.data.paste], { type: "text/plain" })
```
Comment (lem6ns) [inline src/js/injection_script.js]: optional chaining not needed here

File: src/js/injection_script.js [hunk 2]
```diff
@@ -2850,31 +2850,65 @@ ensureDomLoaded(() => {
...
         
                         fetch(`https://publisher.linkvertise.com/api/v1/redirect/link/static${linkvertise_link}?X-Linkvertise-UT=${ut}`).then(r => r.json()).then(json => {
-                            if (json?.data.link.target_type !== 'URL') {
-                                return insertInfoBox('Due to copyright reasons we are not bypassing linkvertise stored content (paste, download etc)');
+                            let target_type = json?.data.link.target_type
+                            const make_serial = () => {
+                                return btoa(JSON.stringify({
+                                    timestamp:new Date().getTime(),
+                                    random:"6548307",
+                                    link_id:json.data.link.id
```
Comment (lem6ns) [inline src/js/injection_script.js]: add spaces after colons

**Review by lem6ns — APPROVED** [review]
lgtm

File: src/js/injection_script.js (hunk 2 above)
Comment (davidmeower) [reply src/js/injection_script.js]: I didn't even notice that myself.

**Comment by davidkra230** [comment]
I will be fixing anything that the reviewers need me to fix and after I have fixed everything, I will squash the commits into one and it will be ready to ship. (I think)

**Review by lostdusty — APPROVED** [review]

File: src/js/injection_script.js [hunk 3]
```diff
@@ -2850,31 +2850,65 @@ ensureDomLoaded(() => {
...
+                                                    )
+                                                }, 3000)
+                                            }
+                                        })
                                     }
-                                })
+                                }
                             }
+                            if (!pathmap.hasOwnProperty(target_type))
+                                return insertInfoBox('Due to copyright reasons we are not bypassing linkvertise stored content (paste, download etc)');
```
Comment (ghost) [inline src/js/injection_script.js]: Rename the sentence to "Due to issues with Linkvertise, we are not bypassing content stored on their site (paste, download, etc)" to make it more clear.

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — CHANGES_REQUESTED** [review]
"Due to legal issues with Linkvertise, we are not bypassing content stored on their site (paste, download, etc)". Can you please change the infobox text to this line?

**Comment by lem6ns** [comment]
> "Due to legal issues with Linkvertise, we are not bypassing content stored on their site (paste, download, etc)". Can you please change the infobox text to this line?

this doesn't really make sense, no? as this PR bypasses their paste system

**Comment by ghost** [comment]
> > "Due to legal issues with Linkvertise, we are not bypassing content stored on their site (paste, download, etc)". Can you please change the infobox text to this line?
> 
> this doesn't really make sense, no? as this PR bypasses their paste system

Ah I see, I was under the impression that we were still were not going to bypass content on their site.

**Review by ghost — APPROVED** [review]

**Pull request merged**